internal/task/gitcheck/check.go

```diff
@@ -63,7 +64,23 @@ func (t Task) Run(ctx *context.Context) error {
 			out = append(out, s.String())
 		}
 
-		return ErrDirty{status: strings.Join(out, "\n")}
+		allowedFiles := ctx.Config.Git.DirtyFiles
```

Comment:
I forgot about this. There is outstanding work to fix the use of the context object within uplift, but I still need to get around to doing it. Since `Git` is optional within the `Config` object, this direct access will cause a panic. Which is why one of the unit tests currently fails.

In the `cmd` package, each command sets up a context by inspecting the config, and command line flags. I would define a `DirtyFiles` field on the `Context` object and set it there i.e. override it with the value from config, if one has been provided. It should then be accessed here via `ctx.DirtyFiles` instead, eliminating the panic. Again this isn't the best, the whole interaction with Cobra needs to be refactored